### PR TITLE
HiFi4 NNLib added as a third party lib for Cadence DSP backend compilation

### PR DESCRIPTION
**Pull request by cad-audio**
HiFi4 NNLib added as a third party lib for Cadence DSP backend compilation

### CODE REVIEW

File: backends/cadence/hifi/kernels/CMakeLists.txt [hunk 1]
```diff
@@ -14,9 +14,10 @@ add_library(
...
-         ${NN_LIB_BASE_DIR}/xa_nnlib/include
-         ${NN_LIB_BASE_DIR}/xa_nnlib/algo/ndsp/hifi4/include/
-         ${NXP_SDK_ROOT_DIR}/middleware/dsp/naturedsp/hifi4/include/
+         ${EXECUTORCH_ROOT}/backends/cadence/hifi/third-party/nnlib/nnlib-hifi4/xa_nnlib/algo/common/include/
+         ${EXECUTORCH_ROOT}/backends/cadence/hifi/third-party/nnlib/nnlib-hifi4/xa_nnlib/include/nnlib
+         ${EXECUTORCH_ROOT}/backends/cadence/hifi/third-party/nnlib/nnlib-hifi4/xa_nnlib/include
+         ${EXECUTORCH_ROOT}/backends/cadence/hifi/third-party/nnlib/nnlib-hifi4/xa_nnlib/algo/ndsp/hifi4/include/
 )
+
+target_link_libraries(cadence_kernels PRIVATE xa_nnlib)
```
Comment (mcremon-meta) [inline backends/cadence/hifi/kernels/CMakeLists.txt:23]: Nit: missing new line will probably make the linter unhappy

File: backends/cadence/hifi/operators/CMakeLists.txt [hunk 2]
```diff
@@ -20,10 +20,10 @@ endif()
...
 # ATen compliant ops that are needed to run this model.
 set(_aten_ops__srcs
-    "${CMAKE_CURRENT_SOURCE_DIR}/op_add.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/op_embedding.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/op_full.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/op_view_copy.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_add.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_embedding.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_full.cpp"
+    "${EXECUTORCH_ROOT}/kernels/portable/cpu/op_view_copy.cpp"
```
Comment (mcremon-meta) [inline backends/cadence/hifi/operators/CMakeLists.txt:26]: Nit: let's move those below the util functions and in alphabetical order. They will eventually be pointing to the optimized versions right? I will remove those 4 ops from the reference folder too and use the portable ops in another PR.

**Review by mcremon-meta — COMMENTED** [review]
Thanks! Looks fine to me minus a couple minor comments. Asked @tarun292 to take a look and make it's fine from the ET side

**Review by tarun292 — APPROVED** [review]

File: backends/cadence/hifi/kernels/CMakeLists.txt (hunk 1 above)
Comment (cad-audio) [reply backends/cadence/hifi/kernels/CMakeLists.txt]: done

File: backends/cadence/hifi/operators/CMakeLists.txt (hunk 2 above)
Comment (cad-audio) [reply backends/cadence/hifi/operators/CMakeLists.txt]: done

**Comment by mcremon-meta** [comment]
@tarun292 can you merge this when you get a chance? I don't seem to have permissions to do that :)

**Review by hsharma35 — APPROVED** [review]

**Review by tarun292 — APPROVED** [review]

**Pull request merged**